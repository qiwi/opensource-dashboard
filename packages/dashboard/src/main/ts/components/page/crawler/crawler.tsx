import { Block, Box, Flex } from '@qiwi/pijma-core'
import { Button, TextField } from '@qiwi/pijma-desktop'
import React, { useState, useEffect, Fragment } from 'react'
import { crawlerService } from '../../../service'
import { useAsync } from '../../../hooks/useAsyncExecutor'

export const Crawler = () => {
  const [getStatus, statusStatus, isRunning] = useAsync(
    crawlerService.getStatus,
  )
  const [stopCrawler, statusStopCrawler] = useAsync(crawlerService.stop)
  const [startCrawler, statusStartCrawler] = useAsync(crawlerService.stop)
  const [secret, setSecret] = useState('')

  useEffect(() => {
    getStatus()
  }, [])

  return (
    <Flex mt={6}>
      <Block css={{ width: '100%', padding: '48px' }}>
        {statusStatus === 'pending' ||
        statusStopCrawler === 'pending' ||
        statusStartCrawler === 'pending' ? (
          <div>Loading...</div>
        ) : (
          <Fragment>
            <Box mb={'24px'} width={64}>
              <TextField
                title={'Secret'}
                type="password"
                value={secret}
                onChange={setSecret}
              />
            </Box>
            <Box width={64}>
              <Button
                type="button"
                kind="brand"
                size="normal"
                text={
                  isRunning === true
                    ? 'Остановить круалинг'
                    : '"Начать крулинг"'
                }
                onClick={
                  isRunning === true
                    ? () => stopCrawler(secret)
                    : () => startCrawler(secret)
                }
              />
            </Box>
          </Fragment>
        )}
      </Block>
    </Flex>
  )
}
