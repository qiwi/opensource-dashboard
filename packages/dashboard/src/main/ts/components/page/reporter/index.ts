export {
  versionsDefault,
  treeDefault,
  usageDefault,
  defaultTypes,
} from './constants'
export { Reporter } from './reporter'
