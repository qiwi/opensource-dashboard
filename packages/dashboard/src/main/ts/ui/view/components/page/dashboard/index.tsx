import React from 'react'

import { DashboardList } from '../../org/dashboardList'
export const Dashboard = () => (
  <div><DashboardList /></div>
)
