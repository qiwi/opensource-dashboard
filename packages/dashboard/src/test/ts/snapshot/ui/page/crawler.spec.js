import React from 'react'
import { PijmaContext } from '../../../pijmaContext'
import { render } from '@testing-library/react'
import { Crawler } from '../../../../../main/ts/components/page/crawler'
import '@testing-library/jest-dom/extend-expect'

describe('Crawler', () => {
  it('renders correctly', () => {
    const { container } = render(
      <PijmaContext>
        <Crawler />
      </PijmaContext>,
    )
    expect(container).toMatchSnapshot()
  })
})
