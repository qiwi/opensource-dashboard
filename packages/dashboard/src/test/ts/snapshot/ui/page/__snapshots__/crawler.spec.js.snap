// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawler renders correctly 1`] = `
<div>
  <div
    class="css-1rwr66l"
  >
    <div
      class="css-4gfk9w"
    >
      <div>
        Loading...
      </div>
    </div>
  </div>
</div>
`;
